### DISCUSSION TITLE
Open a gitter channel

### DISCUSSION THREAD

**Discussion by cornerman** [cloudreach/sceptre#741] [Ideas]
How about we add a gitter channel to this repository?

I think, it would really help to have a channel to discuss problems, ideas, setups or pose questions - basically to share knowledge. I do not know whether there already exists an active community (like reddit, irc, slack) about sceptre, but at least I could not find it (please do point me to it, if there is one).

**Comment by zaro0508** [comment]
Great idea and was also requested in PR https://github.com/cloudreach/sceptre/issues/611

**Comment by ngfgrant** [comment]
Happy for this to be community driven. I checked with our IT team regarding using our Slack. Unfortunately, this won't be possible.

I'd really like to keep the footprint of discussion as small as possible i.e. GitHub and StackOverflow. 

Ideally, if there is an idea for a feature or improvement I'd want to have that discussion here so that there is a clear record of the conversation which is very helpful for maintainers and developers.

For questions I get that GitHub is not the most obvious place but I am happy for questions to be asked here and I just label them as `type: question`. StackOverflow I would say is the defacto place for questions though and we do have our own `sceptre` tag. I guess we need to build up our presence there. 

As for Gitter - more than happy for a channel to be set up if there are a group of people willing to administer it and we can add a link to our readme.

**Comment by ngfgrant** [comment]
We will also benefit from GitHub teams in our new org (officially moving 1st July) https://github.com/orgs/Sceptre/teams/all-contributors

**Comment by cornerman** [comment]
@ngfgrant Cool, I think those great steps to improve community interaction!

@zaro0508 Thanks for opening the IRC channel! I am fine with IRC, but in general, I think that gitter might be more accessible for github users. But let's try :)

**Comment by ngfgrant** [comment]
I definitely still have some reservations about the usefulness of an instant-message and a little apprehensive of useful/helpful question content being lost in chat history vs having a "static" record on StackOverflow/GitHub issue.

One of the things to consider is our community is globally distributed and I don't think IM works that well for multiple timezones. 

We also need to decide _what_ the IM will be for. If it is for general chit chat then that is ok but as soon as it gets into discussions around future of project, current tasks or questions about how to use the tool then I think all of those scenarios are better handled in an asynchronous setting.

From most Gitter channels that I've been part of people tend to just pile in with their problem and then leave or there is multiple conversation threads going on at once. I do think that GitHub Team discussions will be a better way of collating threads of conversation. 

Between the two proposed options I'd agree with @cornerman that Gitter is a bit more "accessible". We could also consider a mailing list via a google group (old school I know)?

**Comment by cornerman** [comment]
@ngfgrant I think, you are raising some good points. It will be impossible to draw a strict line between the instant-messaging channel and the structured discussions/issues in the github organization. Still, I believe a dedicated channel will help newcomers, as it lowers the burden of asking a question. A github issue just feels more serious and you might be reluctant to open one when you just start using the tool or have insufficient information.

When i started using sceptre, I had a lot of questions, because the documentation is not totally complete and therefore I just started looking into the code and trying things out. A simple question answering platform in a channel would have helped me.

Furthermore, I find it interesting to discuss setups on how you could or should use sceptre in certain scenarios. Or questions that are merely related to sceptre but more about the environment that is python, cloudformation or aws. In the end, we are all facing similar problems and challenges. I do not think that these things belong into issues within this repo, they fit perfectly into a chat.

It would be great if we could archive and capture all the relevant information from the discussions. So if there are interesting insides, bugs, feature-ideas or proposals for future development, we should move that part of the discussion into an issue in the github repo. And I think, all people involved are interested in having important stuff tracked, so I hope that transition should work.

**Comment by RobReus** [comment]
It would be really nice to have a place where informal discussions could be held about this project and anything related to it. I do agree that real decision making should not be done via an IM setting. However, that doesn't mean initial ideas could not be coined or discussed via IM first and then move to an issue when it calls for a more formal discussion.

I haven't used GitHub Team Discussions yet, but if there's the wish to keep the footprint small, than using GitHub Team Discussions might be a nice middle ground between GitHub Issues and something like Gitter or Slack.

As a long time user and fan of this project, I would like to do my part and contribute to Sceptre. Having a place where informal discussions could be held would help me do that. Having a place to informally discuss issues or ideas for new features with fellow developers can really speed up the process in my opinion.

**Comment by zaro0508** [comment]
@cornerman @RobReus a #sceptre channel has been created in https://og-aws.slack.com,  join with http://slackhatesthe.cloud  

hoping to see more fans unite.
